app/queries/affected_by_recurring_blackout_query.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class AffectedByRecurringBlackoutQuery < QueryBase
+  def initialize; end
+
+  def call(res_dates)
+    Reservation.where(due_date: res_dates)
```

Comment:
Following up on my other comment, we can replace `Reservation` with `@relation`. We will probably want to add a test to verify that you can pass in a custom relation - you can just pass `Reservation.where.not(id: ID_OF_ONE_OF_THE_MATCHING_RESERVATIONS)` so you can use the same setup as the existing test.